Integrations/GoogleCloudStorage/GoogleCloudStorage.py

```diff
@@ -0,0 +1,566 @@
+
+    demisto.results({
+        "Type": entryTypes["note"],
+        "ContentsFormat": formats["json"],
+        "Contents": result,
+        "HumanReadable": human_readable_table("Buckets in project " + client.project, result),
+        "EntryContext": {ec_key("GCS.Bucket", "Name"): result}
+    })
+
+
+def gcs_get_bucket():
+    bucket_name = demisto.args()["bucket_name"]
+
+    client = init_storage_client()
+    bucket = client.get_bucket(bucket_name)
+    result = bucket2dict(bucket)
+
+    demisto.results({
+        "Type": entryTypes["note"],
+        "ContentsFormat": formats["json"],
+        "Contents": result,
+        "HumanReadable": human_readable_table("Bucket " + bucket_name, result),
+        "EntryContext": {ec_key("GCS.Bucket", "Name"): result}
+    })
+
+
+def gcs_create_bucket():
+    bucket_name = demisto.args()["bucket_name"]
+    bucket_acl = demisto.args().get("bucket_acl", "")
+    default_object_acl = demisto.args().get("default_object_acl", "")
+
+    client = init_storage_client()
+    bucket = client.create_bucket(bucket_name)
+    if bucket_acl:
+        bucket.acl.save_predefined(bucket_acl)
+    if default_object_acl:
+        bucket.default_object_acl.save_predefined(default_object_acl)
+
+    demisto.results({
+        "Type": entryTypes["note"],
+        "ContentsFormat": formats["text"],
+        "Contents": "Bucket {} was created successfully.".format(bucket_name)
+    })
+
+
+def gcs_delete_bucket():
+    bucket_name = demisto.args()["bucket_name"]
+    force = demisto.args().get("force", "") == "True"
+
+    client = init_storage_client()
+    bucket = client.get_bucket(bucket_name)
+    bucket.delete(force)
+
+    demisto.results({
+        "Type": entryTypes["note"],
+        "ContentsFormat": formats["text"],
+        "Contents": "Bucket {} was deleted successfully.".format(bucket_name)
+    })
+
+
+''' Object operations '''
+
+
+def blob2dict(blob):
+    """
+    Converts a google.cloud.storage.Blob to context format (GCS.BucketObject).
+    Note: "blob" is the client API name for what is normally called an "object" in Google Cloud Storage.
+    """
+    return {
+        "Name": blob.name,
+        "Bucket": blob.bucket.name,
+        "ContentType": blob.content_type,
+        "TimeCreated": datetime2str(blob.time_created),
+        "TimeUpdated": datetime2str(blob.updated),
+        "TimeDeleted": datetime2str(blob.time_deleted),
+        "Size": blob.size,
+        "MD5": blob.md5_hash,
+        "OwnerID": "" if not blob.owner else blob.owner.get("entityId", ""),
+        "CRC32c": blob.crc32c,
+        "EncryptionAlgorithm": blob._properties.get("customerEncryption", {}).get("encryptionAlgorithm", ""),
+        "EncryptionKeySHA256": blob._properties.get("customerEncryption", {}).get("keySha256", ""),
+    }
+
+
+def download_blob(blob, file_name=""):
+    cur_directory_path = os.getcwd()
+    file_name = file_name or blob.name.replace("\\", "/").split("/")[-1] or demisto.uniqueFile()
```

Comment:
if `blob.name` is an empty string, this will raise `IndexError`

Reply:
Nope:
`>>> ''.replace('\\', '/').split('/')[-1]`
`''`